frontend/src/pages/accounts/profile.page.tsx

```diff
@@ -373,6 +423,47 @@ const Profile: NextPage = () => {
   const topBanners = allAliases.length > 0 ? banners : null;
   const bottomBanners = allAliases.length === 0 ? banners : null;
 
+  // Render the upsell banner when a user has reached the free mask limit
+  const UpsellBanner = () =>
+    freeMaskLimitReached && isPhonesAvailableInCountry(runtimeData.data) ? (
+      <div className={styles["upsell-banner"]}>
+        <div className={styles["upsell-banner-wrapper"]}>
+          <div className={styles["upsell-banner-content"]}>
+            <p className={styles["upsell-banner-header"]}>
+              {isPhonesAvailableInCountry(runtimeData.data)
+                ? l10n.getString("profile-maxed-aliases-with-phone-header")
+                : l10n.getString("profile-maxed-aliases-without-phone-header")}
+            </p>
+            <p className={styles["upsell-banner-description"]}>
+              {l10n.getString(
+                isPhonesAvailableInCountry(runtimeData.data)
+                  ? "profile-maxed-aliases-with-phone-description"
+                  : "profile-maxed-aliases-without-phone-description",
+                {
+                  limit: freeMaskLimit,
+                },
+              )}
+            </p>
+            <LinkButton
+              href="/premium#pricing"
+              ref={headerBannerSubscriptionLinkRef}
+            >
+              {l10n.getString("profile-maxed-aliases-cta")}
+            </LinkButton>
+          </div>
+          <Image
+            className={styles["upsell-banner-image"]}
+            src={
+              isPhonesAvailableInCountry(runtimeData.data)
```

Comment:
Nice job using this function rather than just checking if the user in the US as the story suggests :)